### PR TITLE
Reorganize working with objects guide, better property definition

### LINKED ISSUE #9419: Issue with "Working with objects": It doesn't cover object injection
MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Working_with_Objects

#### What information was incorrect, unhelpful, or incomplete?
It doesn't talk about object injection, as discussed [here](https://github.com/nodesecurity/eslint-plugin-security/blob/master/docs/the-dangers-of-square-bracket-notation.md)

#### Specific section or headline?

Missing information.

#### What did you expect to see?

A proper solution to sanitize input that will be used to add properties to objects.

#### Did you test this? If so, how?

When using `obj[key] = userValue`, if `key` is any word on the Object prototype (e.g. `constructor`), then any user can modify this object. It seems to be a big security vulnerability and thus needs to be mentioned on this page.

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/web/javascript/guide/working_with_objects`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Working_with_Objects
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/javascript/guide/working_with_objects/index.md
* Last commit: https://github.com/mdn/content/commit/6d4fbc7c53ce9835fa944ed9ddc8866efe51997a
* Document last modified: 2021-09-27T21:01:28.000Z

</details>

### LINKED ISSUE #21340: Poor definition for understanding what a property is.
### MDN URL

https://developer.mozilla.org/en-US/docs/Glossary/property/JavaScript

### What specific section or headline is this issue about?

definition

### What information was incorrect, unhelpful, or incomplete?

> A JavaScript property is a characteristic of an object, often describing attributes associated with a data structure.

This tells me that properties are associated with objects. However, the definition of a characteristic is not very exact. From Oxford:

a feature or quality belonging typically to a person, place, or thing and serving to identify it.

A feature? This still doesn't tell me what a property _**IS**_. The definition says that it often (note: not always) describes attributes associated with a data structure. Which data structure? Specifically objects? Since properties are associated with objects? If just objects, why not say so?

All this ambiguity is well and good if it were just general terms that have no meaning, but there are actual code methods like hasOwnProperty() which refers me back to the entry for property. I have to understand hasOwnProperty() so I have to understand property which I can't because the definition given is too vague to make any sense.

### What did you expect to see?

A firm definition of the word property as used in JavaScript so that a beginner can understand the term. Knowing what a property EXACTLY is is important to the knowledge of programmers.

### Do you have any supporting links, references, or citations?

No, just the actual MDN page.

### Do you have anything more you want to share?

I count on MDN as my source for JavaScript knowledge and I think they do a better job than ANY with JavaScript. I value the free resource and it has furthered my learning so much. I wish they did every language. I want to improve the site giving them a new programmer's viewpoint.

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/glossary/property/javascript`
* MDN URL: https://developer.mozilla.org/en-US/docs/Glossary/property/JavaScript
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/glossary/property/javascript/index.md
* Last commit: https://github.com/mdn/content/commit/ed947b2c608428b62a60f07d09dc543f732dc09b
* Document last modified: 2022-09-21T02:09:26.000Z

</details>

### PR DESCRIPTION
**Pull request by Josh-Cena**
### Description

Fix https://github.com/mdn/content/issues/9419
Fix https://github.com/mdn/content/issues/21340

### CODE REVIEW

File: files/en-us/glossary/property/javascript/index.md [hunk 1]
```diff
@@ -5,18 +5,36 @@ tags:
...
+[Accessor properties](/en-US/docs/Web/JavaScript/Data_structures#accessor_property) do not have an actual "value". The value is represented indirectly through a pair of functions, one (the getter) invoked when reading the value and one (the setter) invoked when setting the value. However, accessor properties behave like regular data properties on the surface, because the getter and setter functions are invoked automatically and are typically transparent to JavaScript code.
 
-> **Note:** When we say that "a property holds an object", that is shorthand for "a property holds an object _reference_". This distinction matters because the original referenced object remains unchanged when you change the property's value.
+The property's value (including the getter and setter) and its attributes are stored in a data record called the _property descriptor_. Many methods, such as {{jsxref("Object.getOwnPropertyDescriptor()")}} and {{jsxref("Object.defineProperty()")}}, work with property descriptors.
+
+The term _property_ itself does not correspond to any JavaScript value — it's an abstract concept. For example, in the following code:
+
```
Comment (estelle) [inline files/en-us/glossary/property/javascript/index.md:21]:
```suggestion

```js
const obj = {
  a: 1,
  b: function() {},
};
```js

which can be written more simply as:

```

As an old-school programmer, I had never seen this simpler syntax. This makes it clearer to me, but possibly this has been around for years and I am one of the few who hadn't seen it?

**Review by estelle — APPROVED** [review]
One suggested optional addition...

File: files/en-us/glossary/property/javascript/index.md (hunk 1 above)
Comment (Josh-Cena) [reply files/en-us/glossary/property/javascript/index.md:21]: MDN has switched to the method syntax everywhere, and in the near future there's very likely linter rules to catch this. I was specifically trying to illustrate how methods are also properties, so I figured maybe using the method syntax makes that more apparent. If this is potentially confusing, I may use a normal non-function value instead.

Comment (estelle) [reply files/en-us/glossary/property/javascript/index.md:21]: My take is that this is a good opportunity to teach people this new method. But i am not as familiar with the JS docs as the rest of the OWD team

File: files/en-us/glossary/property/javascript/index.md [hunk 2]
```diff
@@ -5,18 +5,36 @@ tags:
...
+```js
+const obj = {
+  a: 1,
+  b() {},
+};
+```
+
+The object `obj` has two properties. The first one has `"a"` as the key and `1` as the value. The second one has `"b"` as the key and a function as the value. The `"a"` – `1`, `"b"` – `function` associations are the properties of the object.
```
Comment (Josh-Cena) [inline files/en-us/glossary/property/javascript/index.md:29]: What about this:

```suggestion
The object `obj` has two properties. The first one has `"a"` as the key and `1` as the value. The second one has `"b"` as the key and a function as the value (using the [method syntax](/en-US/docs/Web/JavaScript/Reference/Functions/Method_definitions)). The `"a"` – `1`, `"b"` – `function` associations are the properties of the object.
```

Comment (estelle) [reply files/en-us/glossary/property/javascript/index.md:29]: Yes!

**Pull request merged**